driver-core/src/main/com/mongodb/internal/operation/AsyncQueryBatchCursor.java

```diff
@@ -337,7 +338,17 @@ private void endOperationInProgress() {
     private void handleGetMoreQueryResult(final AsyncConnection connection, final SingleResultCallback<List<T>> callback,
                                           final QueryResult<T> result, final boolean tryNext) {
         cursor.set(result.getCursor());
-        if (!tryNext && result.getResults().isEmpty() && result.getCursor() != null) {
+        if (isClosePending) {
+            try {
+                connection.release();
+                if (result.getCursor() == null) {
+                    connectionSource.release();
+                }
+                endOperationInProgress();
+            } finally {
+                callback.onResult(null, null);
```

Comment:
Consider whether we should report an error here instead of indicating that the cursor has been exhausted normally.

Reply:
Yes, we may either return null, thus signalling `onComplete` or throw an exception, thus signalling `onError`. According to https://github.com/reactive-streams/reactive-streams-jvm/blob/v1.0.2/README.md#user-content-1.6 and https://github.com/reactive-streams/reactive-streams-jvm/blob/v1.0.2/README.md#user-content-2.4 both options are valid, but `onComplete` seems more natural to me: a user requested to cancel the subscription, and we cancelled it - there is nothing unexpected or exceptional about this situation.

The problem is that Reactor calls neither `onError` nor `onComplete` despite `AsyncChangeStreamBatchCursor.convertResultsCallback` passing the result/exception to the callback provided by the `MongoOperationPublisher.sinkToCallback` method. This is not related to the changes in this MR and may be a manifestation of us not hooking up to Reactor properly.

Reply:
I dont think you need to signal anything back to the subscriber - so reactor not calling onError / onComplete post cancellation is correct.

> If a Subscription is cancelled its Subscriber MUST eventually stop being signaled.

Reactor may stop signals reaching the subscriber straight way.

Reply:
So for me `callback.onResult(null, null)` seems fine.